Core/clim-basic/utils.lisp

```diff
@@ -538,20 +538,47 @@ STREAM in the direction DIRECTION."
                     (:vertical   (values 0 1))))
               (/ value (sqrt (+ (* dx dx) (* dy dy))))))))))))
 
+(defun valid-margin-spec-p (margins)
```

Comment:
I would squash the following commits into one:
* `b7ff860b5 margin-spec: move utility to utils and validate the value precisely`
* `dec980b88 parse-space: add corresponding type and enchance the implementation` (typo in "enchance")
* `4b66aacd4 text-formatting: margin-spec-p: do not assume value to be a number`

Reply:
I'd rather keep parse-space type in a separate commit, the other two are squashed (with moved tests).